src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -1048,11 +1048,9 @@ async function main(): Promise<void> {
         case "cloudevent":
           const rawBody = (req as RequestWithRawBody).rawBody;
           let reqBody = JSON.parse(rawBody.toString());
-          if (req.headers["content-type"]?.includes("cloudevent")) {
-            if (EventUtils.isBinaryCloudEvent(req)) {
-              reqBody = EventUtils.extractBinaryCloudEventContext(req);
-              reqBody.data = req.body;
-            }
+          if (EventUtils.isBinaryCloudEvent(req)) {
+            reqBody = EventUtils.extractBinaryCloudEventContext(req);
+            reqBody.data = req.body;
```

Comment:
I wonder if we actually wanted a third option: check for contenttype to include cloudevent but then throw that structured encoding is not supported if isBinaryCloudEvent is false.

Reply:
We are actually removing the check for content type to include cloudevent since it's not in the spec - https://cloud.google.com/eventarc/docs/workflows/cloudevents#payload-format